SparseGrids/tsgCoreOneDimensional.cpp

```diff
@@ -392,6 +393,72 @@ const char* OneDimensionalMeta::getHumanString(TypeOneDRule rule){
     }
 }
 
+// Generate the n roots of the n-th degree orthogonal polynomial.
+// ref_points and ref_points are used in the computation of the integrals that form the elements of the Jacobi matrix.
+double poly_eval(int nroots, std::vector<double> roots, double x) {
+    double eval_value = 1.0;
+    for (int i=0; i<nroots; i++) {
+        eval_value *= (x - roots[i]);
+    }
+    return(eval_value);
+}
+std::vector<std::vector<double>> OneDimensionalOrthPolynomials::getRootCache(
+        int n,
+        std::function<double(double)> &weight_fn,
+        const std::vector<double> &ref_weights,
+        const std::vector<double> &ref_points) {
+
+    // Initialize and compute the integral weights corresponding to weight_fn.
+    std::vector<std::vector<double>> root_cache(n);
+    std::vector<double> roots;
+    #ifdef Tasmanian_ENABLE_BLAS
+    if (ref_points.size() != ref_weights.size()) {
+        throw std::invalid_argument("The number of reference points does not match the point of reference weights!\n");
+    }
+    int nref = ref_points.size();
+    std::vector<double> integral_weights(nref);
+    for (int j=0; j<nref; j++) {
+        integral_weights[j] = weight_fn(ref_points[j]) * ref_weights[j];
+    }
+    // Compute the roots incrementally.
+    std::vector<double> poly_m1_vals(nref, 0.0), poly_vals(nref, 1.0);
+    std::vector<double> alpha(n), beta(n-1), offdiag(n-1);
+    double alpha_numr, alpha_denm, beta_numr, beta_denm;
```

Comment:
In general, do not leave uninitialized variables.

Move the declaration of these inside the for-loop.

Reply:
I moved all of the doubles and ``offdiag`` inside the for-loop, but ``alpha`` and ``beta`` need to be initialized with zero length outside of the for-loop now.